### PR TITLE
Switch Apex with Pytorch

### PR DESCRIPTION
**Pull request by blisc**
- [x] Switch DistributedDataParallel
- [x] Switch SyncBatchNorm
- [x] Address gradient_predivide_factor
- [x] Address gradient accumulation

This PR switches from apex's DistributedDataParallel to torch's DistributedDataParallel.
Warning: that gradient_predivide_factor is no longer working after this switch
Warning: in multi-gpu runs, neural modules with no weights MUST inherit from NonTrainableNM.

### CODE REVIEW

**Comment by blisc** [comment]
For some reason, memory usage shoots up from 12GB -> 20GB, need to find out why

File: nemo/backends/pytorch/actions.py [hunk 1]
```diff
@@ -25,9 +27,8 @@
 
 # these imports will happen on as-needed basis
 amp = None
-convert_syncbn = None
-create_syncbn_process_group = None
-DDP = None
+# convert_syncbn = None
```
Comment (okuchaiev) [inline nemo/backends/pytorch/actions.py]: @redoctopus are you training base models on multi-node setup with sync batch norm?

Comment (redoctopus) [reply nemo/backends/pytorch/actions.py]: Not currently.

Comment (blisc) [reply nemo/backends/pytorch/actions.py]: Sync batchnorm is still supported.
